### PR TITLE
feat(component): `<PricingTiers />`

### PR DESCRIPTION
**Pull request by pbortnick**
🎟️ [Asana Task](https://app.asana.com/0/1200531701094631/1202533326661067/f)
🔍 [Preview Link](https://react-components-git-pbtier-card-hashicorp.vercel.app/components/pricingtiers)
🎨 [Design](https://www.figma.com/file/uxnWWpM14UcttJNvBadRfL/Pricing?node-id=202%3A6335)

---

## Description

Add `<PricingTiers />` component package to be used in new pricing pages project

### PR Checklist 🚀

Items in this checklist may not may not apply to your PR, but please consider each item carefully.

- [x] Add Asana and Preview links above.
- [x] Conduct thorough self-review.
- [x] Add or update tests as appropriate.
- [x] Conduct reasonable cross browser testing for both compatibility and responsive behavior (We have a [Sauce Labs](https://app.saucelabs.com/) account for this, if you don't have access, just ask!).
- [x] Conduct reasonable accessibility review (use the [WAS](https://accessible.org/Web-Accessibility-Standards-WAS-2.pdf) as a guide or an [axe browser plugin](https://www.deque.com/axe/) until we establish more formal checks).
- [x] Identify (in the description above) and document (add Asana tasks on [this board](https://app.asana.com/0/1100423001970639/list)) any technical debt that you're aware of, but are not addressing as part of this PR.

### CODE REVIEW

File: packages/pricing-tier-list/partials/pricing-tier-item/index.tsx [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+      {icon && <div className={s.icon}>{icon}</div>}
+      <h3 className={s.tierName}>{title}</h3>
+      <div className={s.details}>
+        {label && <span className={s.label}>{label}</span>}
+        {price && <span className={s.price}>{price}</span>}
+        {consumption && <span className={s.consumption}>{consumption}</span>}
+      </div>
+      <div
+        className={s.description}
+        dangerouslySetInnerHTML={{ __html: description }}
```
Comment (BRKalow) [inline packages/pricing-tier-list/partials/pricing-tier-item/index.tsx:29]: Do we need to use dangerouslySetInnerHTML here to support the fields coming from Dato as HTML?

File: packages/pricing-tier-list/index.test.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
+import { render, cleanup } from '@testing-library/react'
+import PricingTierList from '.'
+import props from './fixtures/data.json'
+
+const { oneTier, twoTiers, threeTiers, fourTiers, fiveTiers } = props
+
+afterEach(cleanup)
+
+describe('<PricingTierList />', () => {
```
Comment (BRKalow) [inline packages/pricing-tier-list/index.test.js:9]: 👏

File: packages/pricing-tier-list/README.md [hunk 3]
```diff
@@ -0,0 +1,3 @@
+# Pricing Tier List
+
+## @hashicorp/react-pricing-tier-list
```
Comment (BRKalow) [inline packages/pricing-tier-list/README.md:3]: A short description here would be great! Even just one or two sentences 👍

File: packages/pricing-tier-list/partials/pricing-tier-item/style.module.css [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+  width: 100%;
+
+  & p {
+    margin: 0;
+  }
+
+  & a {
+    color: var(--wpl-neutral-300);
+    cursor: pointer;
+  }
```
Comment (BRKalow) [inline packages/pricing-tier-list/partials/pricing-tier-item/style.module.css:157]: Should links in the supplementary content have an underline? I'm seeing that in the Figma. 💭

**Review by BRKalow — COMMENTED** [review]
Looks good overall. Checked out the www preview and all seems pretty solid to me.

File: packages/pricing-tier-list/partials/pricing-tier-item/index.tsx (hunk 1 above)
Comment (pbortnick) [reply packages/pricing-tier-list/partials/pricing-tier-item/index.tsx:29]: Yeah exactly

File: packages/pricing-tier-list/README.md (hunk 3 above)
Comment (pbortnick) [reply packages/pricing-tier-list/README.md]: 👍  Added [here](https://github.com/hashicorp/react-components/pull/644/commits/b2138260c7d8ced2a74b1515b36694b8c6c7f63b). Open to suggestions tho

**Review by alexcarpenter — COMMENTED** [review]
Is there a possibility that consumption isn't set on one of the items causing this alignment issue with the description? 

CleanShot 2022-07-12 at 10 45 37@2x

File: packages/pricing-tier-list/docs.mdx [hunk 5]
```diff
@@ -0,0 +1,246 @@
...
+        label: 'Starting at',
+        price: '$1.578',
+        consumption: 'per hour',
+        description: '<p>Clusters designed to scale with the demand of running production workloads.</p>',
+        cta: {
+          type: 'textLink',
+          url: '#',
+          title: 'Try it for free',
+        },
+        supplementaryInfo: '<p>*Price scales with <a>active clients</a></p>'
```
Comment (alexcarpenter) [inline packages/pricing-tier-list/docs.mdx:108]: Is it not possible to add more than one line item here in the supplementaryInfo? The use of absolute positioning could lead to layout issues if the text isn't the perfect length shown here in the preview. 
CleanShot 2022-07-12 at 10 51 04@2x

**Comment by pbortnick** [comment]
Marking as draft until we are ready to merge later on in the pricing table project

**Review by alexcarpenter — APPROVED** [review]
Looks good, nice work here!

**Pull request merged**